examples/hosts/node-host/src/nodeCodeloader.ts

```diff
@@ -50,26 +51,38 @@ export class NodeCodeLoader {
         const codeEntrypoint = `${packageDirectory}/node_modules/${scope}/${name}`;
 
         // Install the versioned package if not already present.
-        if (!fs.existsSync(packageDirectory)) {
-            fs.mkdirSync(packageDirectory, { recursive: true });
-
-            // Copy over the root .npmrc (if present) to the directory where npm install will be executed.
-            if (fs.existsSync(`${this.packageDirectory}/.npmrc`)) {
-                fs.copyFileSync(`${this.packageDirectory}/.npmrc`, `${packageDirectory}/.npmrc`);
-            }
-
-            // Run npm install
-            await asyncExec(`npm install ${pkg}`, { cwd: packageDirectory });
+        if (!fs.existsSync(signalPath) || !fs.existsSync(codeEntrypoint)) {
+            const pkgDirExists = fs.existsSync(packageDirectory);
+            if (!pkgDirExists || this.useLocalDirectory) {
+                if (!pkgDirExists) {
+                    fs.mkdirSync(packageDirectory, { recursive: true });
+                }
+                // Copy over the root .npmrc (if present) to the directory where npm install will be executed.
+                if (this.useLocalDirectory) {
+                    if (fs.existsSync(`${__dirname}/../.npmrc`)) {
+                        fs.copyFileSync(`${__dirname}/../.npmrc`, `${packageDirectory}/.npmrc`);
+                    }
+                } else if (fs.existsSync(`${this.packageDirectory}/.npmrc`)) {
+                        fs.copyFileSync(`${this.packageDirectory}/.npmrc`, `${packageDirectory}/.npmrc`);
+                }
```

Comment:
Actually can you publish the key-value package to npm first? This copying npmrc is only needed cause we are pulling from a private registry. If the package is already public, we can just remove this and update the readme as well.